src/main/java/org/apache/maven/model/jdom/util/JDomUtils.java

```diff
@@ -61,14 +62,29 @@ public static void addElement(Element element, Element root) {
    */
   public static void addElement(Element element, Element root, int index) {
     root.addContent(
-            index == -1 ? root.getContentSize() - 1 : index,
+            index,
             asList(
                     new Text("\n" + detectIndentation(root)),
                     element));
     resetIndentations(root, detectIndentation(root));
     resetIndentations(element, detectIndentation(root) + "  ");
   }
 
+  public static int getElementIndex(Element element, Element root) {
+    return root.indexOf(element);
+  }
+
+  private static int getLastElementIndex(Element root) {
```

Comment:
I would prefer: `getIndexOfLastElement` and `getIndexOfElement`

Reply:
I think either is OK. isn't it a bit fussy?

Reply:
It's a code review and this was just a suggestion. So, no: I don't think fussy. If you don't like it OKs.